sycl/doc/extensions/proposed/sycl_ext_oneapi_kernel_arg_restrict_property.asciidoc

```diff
@@ -0,0 +1,187 @@
+= sycl_ext_oneapi_kernel_arg_restrict_property
```

Comment:
I wonder if this should be it's own extension or if it should be the starting point for an extension that defines a bunch of kernel argument properties.  For example, we have [sycl_ext_oneapi_kernel_properties ](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_kernel_properties.asciidoc)which defines a collection of 4 kernel properties.  Maybe this should be named "sycl_ext_oneapi_kernel_arg_properties" instead?

Reply:
That could be a good idea. I have left it as is for now. If any other oneapi kernel argument properties arise we can rename and reuse this extension.

Reply:
Once we implement and support this extension it will be harder to change it's name.  If we plan to leave it in the "proposed"  state for a while, we can wait and see if there are other kernel argument properties.

Reply:
Updated. It is easier to do it now and I don't see cases where a vendor may want to support other kernel argument properties that can go in this extension and not support `restrict`.